root/static/scripts/account/components/EditProfileForm.js

```diff
@@ -0,0 +1,334 @@
+      return newState;
+    });
+  }
+
+  addLanguage: () => void;
+
+  addLanguage() {
+    this.setState(function (prevState) {
+      const languageIndex = prevState.form.field.languages.field.length;
+
+      let idMax = 0;
+      const findMax = function rec(prop) {
+        const propKeys = keys(prop);
+        if (propKeys) {
+          if (propKeys.indexOf('id') > -1) {
+            // $FlowFixMe - property id is missing in EditorProfileFormT
+            idMax = idMax < prop.id ? prop.id : idMax;
+          }
+          if (propKeys.indexOf('field') > -1) {
+            // $FlowFixMe - property field is missing in EditorProfileFormT
+            keys(prop.field).forEach((key) => rec(prop.field[key]));
+          }
+        }
+      };
+      findMax(prevState.form.field);
+
+      const newState = {form: prevState.form};
+      // $FlowFixMe - property push is missing in $ReadOnlyArray
+      newState.form.field.languages.field.push({
+        errors: [],
+        field: {
+          fluency: {
+            errors: [],
+            has_errors: false,
+            html_name: 'profile.languages.' + languageIndex + '.fluency',
+            id: idMax + 3,
+            value: undefined,
+          },
+          language_id: {
+            errors: [],
+            has_errors: false,
+            html_name: 'profile.languages.' + languageIndex + '.language_id',
+            id: idMax + 2,
+            value: undefined,
+          },
+        },
+        has_errors: false,
+        html_name: 'profile.languages.' + languageIndex,
+        id: idMax + 1,
+      });
+      return newState;
+    });
+  }
+
+  render() {
+    const field = this.state.form.field;
+    return (
+      <form id="edit-profile-form" method="post">
+        <input
+          hidden
+          id={'id-' + field.username.html_name}
+          name={field.username.html_name}
+          readOnly
+          value={field.username.value}
+        />
+        {DBDefs.STAGING_TESTING_FEATURES ? (
+          <Warning
+            message={l('This is a development server. Your email address is not private or secure. Proceed with caution!')}
+          />
+        ) : null}
+
+        <FormRowEmailLong
+          field={field.email}
+          label={addColon(l('Email'))}
+        />
+        <FormRow hasNoLabel>
+          {l('If you change your email address, you will be required to verify it.')}
+        </FormRow>
+
+        <FormRowURLLong
+          field={field.website}
+          label={addColon(l('Website'))}
+        />
+
+        <FormRowSelect
+          allowEmpty={true}
+          field={field.gender_id}
+          label={l('Gender:')}
+          onChange={this.handleGenderChange}
+          options={genderOptions}
+        />
+
+        <FormRow>
+          <FormLabel
+            forField={field.area.field.name}
+            label={l('Location:')}
+          />
+          <span className="area autocomplete">
```

Comment:
Can you use root/static/scripts/common/components/Autocomplete.js here? It should handle all the messy knockout integration.

Reply:
Return the following error message, probably related to hydration:
```
ReferenceError: window is not defined
    at Object.<anonymous> (/musicbrainz-server/root/static/lib/jquery-ui.js:2:1)
    ...
    at Object.require (/musicbrainz-server/root/static/scripts/common/MB/Control/Autocomplete.js:25:1)
```

Reply:
root/static/scripts/common/components/Autocomplete.js performs `require('../MB/Control/Autocomplete')` at the top-level of the file, whereas it probably needs to be changed to do that in `componentDidMount` instead, so that it doesn't try to run/initialize any jQuery UI code on the server.

Reply:
Changed in https://github.com/metabrainz/musicbrainz-server/pull/730/commits/529e4e568565938ef619c5a680f30d658d443a2b.

`TypeError: subscription.callback is not a function` is triggered on hydration through:
https://github.com/metabrainz/musicbrainz-server/blob/529e4e568565938ef619c5a680f30d658d443a2b/root/static/scripts/common/components/Autocomplete.js#L26